contentcuration/contentcuration/middleware/locale.py

```diff
@@ -15,6 +17,8 @@ def _is_exempt(self, obj):
     def process_view(self, request, callback, callback_args, callback_kwargs):
         if self._is_exempt(callback):
             setattr(request, LOCALE_EXEMPT, True)
+            translation.activate(settings.LANGUAGE_CODE)
+            request.LANGUAGE_CODE = translation.get_language()
```

Comment:
I think the actual issue here is that we are not overriding the `process_request` method of the `LocaleMiddleware`: https://github.com/django/django/blob/stable/3.2.x/django/middleware/locale.py#L18

This fixes it to some extent by overriding this, but I think we should follow the same pattern for the `process_request` as we do for the `process_response` method. Checking if this view is exempt, and in that case not invoking the super call.

If you could test this alternative approach and see if it fixes it, that would feel more robust.

Reply:
Yeah, I feel you sir. I was of the same opinion that overriding `process_request` would be more robust. But the problem is `process_request` runs before Django knows what view to run, so we don't know the View that's going to be run in `process_request`. 

That's why I chose `process_view`. And the current code IMO is robust as well, I don't see any particular concern sir. 

Let me know your thoughts sir.

Reply:
Yeah!